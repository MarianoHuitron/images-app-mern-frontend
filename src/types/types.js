

export const types = {

    uiSetError: '[UI] Set Error Form',
    uiRemoveError: '[UI] Remove Error Form',
    uiSetLoading: '[UI] Set Loading',
    uiRemoveLoading: '[UI] Remove Loading',

    authCheckingFinish: '[AUTH] finish checking login state',
    authStartLogin: '[AUTH] Start login',
    authStartRegister: '[AUTH] Start register',
    authStartTokenRenew: '[AUTH] Start Renew Token',
    authLogin: '[AUTH] login',
    authRegister: '[AUTH] register',
    authLogout: '[AUTH] logout',

    imageStartUpload: '[IMG] Start Upload',
    imageStartLoading: '[IMG] Start uploading image loading',
    imageStartDeleted: '[IMG] Start deleted image',
    imageDeleted: '[IMG] Image deleted',
    imageUploaded: '[IMG] uploaded',
    imageStartGet: '[IMG] Start Get Images',
    imageGet: '[IMG] Get Images',
    imageFilter: '[IMG] Apply Filter',
    imagesLogout: '[IMG] Logout Images'
}