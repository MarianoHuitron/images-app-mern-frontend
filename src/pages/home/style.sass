img 
    max-width: 100%
    display: block



.masonry-gallery
    columns: 6 300px
    column-gap: 1rem

    .masonry-gallery-item
        // width: 150px
        width: 100%
        margin: 0 1rem 1rem 0

        &:hover
            cursor: pointer

        .is-selected
            filter: brightness(40%)

.delete-btn
    border-radius: 100%
    padding: 1rem 1.2rem
    color: white
    right: 2%
    bottom: 5%
    &:focus
        outline: none