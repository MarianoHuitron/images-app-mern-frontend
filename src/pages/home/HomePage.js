import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import './style.sass';
import { Link } from 'react-router-dom';
import { startDeleteImage, startGetImages } from '../../actions/images';

export const HomePage = () => {

    const dispatch = useDispatch();
    const { images_filtered: images } = useSelector(state => state.images)
    const [isSelected, setIsSelected] = useState(null)

    useEffect( () => {
        dispatch( startGetImages() );
    }, [dispatch])

    const handleSelecteImage = (id) => {
        if(isSelected && isSelected === id) {
            setIsSelected(null)
        } else {
            setIsSelected(id)
        }
    }

    const handleDeleteImage = () => {
        dispatch( startDeleteImage( isSelected ) );
    } 

    return (
        <div className="w-full flex py-24 mx-auto px-2 md:px-12">
            
            <div className="masonry-gallery mx-auto ">
                {
                    images.map((img) => (
                        <div 
                            key={img.asset_id} 
                            className="masonry-gallery-item relative"
                            onClick={ () => handleSelecteImage(img._id) }
                        > 
                            <img 
                                src={img.img_path} 
                                alt={img.title} 
                                className={ `rounded  ${(isSelected === img._id) && 'is-selected'}` } /> 

                            {
                                (isSelected && img._id === isSelected) 
                                    &&
                                <button
                                    className="bg-red-600 text-white h-8 w-8 absolute rounded-full cursor-pointer focus:outline-none"
                                    style={{top: '2%', right: '2%'}}
                                    onClick={ handleDeleteImage }
                                > 
                                    <i className="fas fa-trash"></i>
                                </button>                            
                            }
                                
                        </div>
                    ))
                }
            </div>
            <Link to="/upload" className="bg-red-500  fixed bottom-0 delete-btn"> 
                <i className="fas fa-plus"></i>
            </Link>
        </div>
    )
}
