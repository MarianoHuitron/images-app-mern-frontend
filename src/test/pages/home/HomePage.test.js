import React from 'react';
import { mount } from "enzyme"
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk' 
import { HomePage } from '../../../pages/home/HomePage';
import { Provider } from 'react-redux';
import { types } from '../../../types/types';

const middleware = [thunk];
const mockStore = configureStore(middleware);
const initState = {
    auth: {
        checking: false,
        user: {
            uid: '123',
            name: 'jose'
        }
    }
};
const store = mockStore(initState);

const wrapper = mount(
    <Provider store={ store } >
        <HomePage />
    </Provider>
);

describe('Testing <HomePagr />', () => {

    test('should to show the <HomePage />', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should to logout', () => {
        wrapper.find('button').prop('onClick')();
        const actions = store.getActions();
        expect( actions[0] ).toEqual({type: types.authLogout})
    })
    
    

})

