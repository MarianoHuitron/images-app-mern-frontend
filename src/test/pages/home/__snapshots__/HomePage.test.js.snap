// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HomePagr /> should to show the <HomePage /> 1`] = `
<div
  className="w-full  flex"
>
  <div
    className="mx-auto pt-12 animate__animated animate__fadeIn"
  >
    <img
      alt="welcome"
      className="text-center h-56"
      src="./assets/welcome.svg"
    />
    <div
      className="text-center mt-10"
    >
      <button
        className="py-3 bg-green-500 text-white w-full hover:bg-green-600 cursor-pointer rounded outline-none"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
